lib/bibliothecary/parsers/npm.rb

```diff
@@ -149,6 +149,18 @@ def self.lockfile_preference_order(file_infos)
           ] + transform_tree_to_array(metadata.fetch("dependencies", {}))
         end.flatten(1)
       end
+
+      private_class_method def self.parse_name_from_package_lock_v2(name)
```

Comment:
🎱 Any particular reason for the `_v2` naming?

Reply:
I'm working on a round of changes based on @tiegz's feedback above, and this method will go away actually. But the v2 naming here refers to the lockfile version